client/analytics/settings/index.js

```diff
@@ -116,6 +116,9 @@ class Settings extends Component {
 	 * @param {object} state - State
 	 */
 	persistChanges( state ) {
+		// TODO: Should remove global state from the file. This creates
+		// potential hard to debug side-effects.
```

Comment:
If I understood this correctly, we can't remove this until minimum WooCommerce version is increased, right? Should we explain that in the comment?

Reply:
No, we can remove this in further refactors after this is merged and replace with usage of the new `setSetting` interface.  While `setSetting` and `getSetting` still interact with a global under the hood, it hides it behind an interface so that it's more controlled.  I mostly added this comment just as a reminder to address it.

Reply:
see https://github.com/woocommerce/woocommerce-admin/issues/2943